dxp/aui_script.md

```diff
@@ -0,0 +1,148 @@
+
+### What it does
+
+This tag allow users to use and require ES module/resources inside a JSP.
+
+We have three possible use cases when using this tag:
+
+1. Inline JS on the JSP
+2. Import an AUI/YUI module
+3. Import a ES module
+
+#### Inline JS on the JSP
+
+By just placing JavaScript code inside an `aui:script` tag, you can take advantage of optimizing scripts and place them to be made available in the final compressed JavaScript bundle. It can be achieved when appending the script to the [`AUI_SCRIPT_DATA` WebKey](https://github.com/liferay/liferay-portal/blob/67b569099146a4b999e2fad7d7d1a9794a337f0b/portal-kernel/src/com/liferay/portal/kernel/util/WebKeys.java#L55) and all these scripts will be applied at the [bottom of the page](https://github.com/liferay/liferay-portal/blob/e3ffac158e0ec5acc5c67069fbd7ba688d3c78d4/portal-web/docroot/html/common/themes/bottom.jsp#L44). Finally the [Aggregate Filter will take care](https://github.com/liferay/liferay-frontend-guidelines/blob/bc6dae8514af04a6384a7bea9d4ddf266087ffc5/dxp/resource_serving.md#aggregate-filter), of creating the final bundle with all the JavaScript on the page. It is worth mentioning that the same rule applies to all the following use cases.
+
+Like this example:
+
+```jsp
+<aui:script>
+	Liferay.fire('myGlobalEvent', {
+		payload: 'myPayload',
+	});
+</aui:script>
+```
+
+it will be turned on:
+
+```js
+Liferay.fire('myGlobalEvent', {
+	payload: 'myPayload'
+});
+```
+
+#### Import an AUI/YUI module using the `use` property
+
+> ⚠️ This is a deprecated use case, though it still works if someone configures it.⚠️
+
+Each AUI / YUI module listed as a value for the `use` property will be [loaded asyncronously by the `AUI().use` function](https://github.com/yui/yui3/blob/master/src/yui/js/yui.js#L1323..L1331).
+
+You can import several YUI / AUI modules within the same `use` property, as in this example:
+
+Here is an example:
+
+```jsp
+<aui:script use="liferay-alert">
+	new Liferay.Alert(
+		{
+			closeable: true,
+			message: 'I like drink tea while eating french potatoes',
+			type: 'success'
+		}
+	).render('#myId');
+</aui:script>
+```
+
+The output will be a code like this:
+
+```js
+AUI().use('liferay-alert', function (A) {
+	(function() {var $ = AUI.$;var _ = AUI._;
+		new Liferay.Alert(
+			{
+				closeable: true,
+				message: 'I like drink tea while eating french potatoes',
+				type: 'success'
+			}
+		).render('#myId');
+	})()
+}
+```
+
+Note in the AlloyUI components, coming from the alloy-ui repository, classes are made available from the global variable `A` on the scope of the `aui:script`. Here as in this [code snippet the Menu component is defined](https://github.com/liferay/alloy-ui/blob/master/src/aui-menu/js/aui-menu.js#L39]).
```

Comment:
I would change:

> on the scope of the `aui:script`. Here as in this

to:

> on the scope of the `aui:script` (as in this

If I understood the sentence correctly...

Reply:
Makes sense :) Updating